easybuild/easyconfigs/b/bcl2fastq2/bcl2fastq2-2.20.0-foss-2016b-Python-2.7.12.eb

```diff
@@ -0,0 +1,37 @@
+# This file is an EasyBuild reciPY as per https://github.com/easybuilders/easybuild
+
+easyblock = 'ConfigureMake'
+
+name = 'bcl2fastq2'
+version = '2.20.0'
+versionsuffix = '-Python-%(pyver)s'
+
+homepage = 'https://support.illumina.com/sequencing/sequencing_software/bcl2fastq-conversion-software.html'
+description = """bcl2fastq Conversion Software both demultiplexes data and converts BCL files generated by
+ Illumina sequencing systems to standard FASTQ file formats for downstream analysis."""
+
+toolchain = {'name': 'foss', 'version': '2016b'}
+toolchainopts = {'pic': True}
+
+source_urls = ['ftp://webdata2:webdata2@ussd-ftp.illumina.com/downloads/software/bcl2fastq/']
+sources = [{
+    'filename': '%s-v%s-tar.zip' % (name, version.replace('.', '-')),
+    'extract_cmd': 'unzip -p %s | tar -xzvf -',  # source file is a .zip that contains a .tar.gz
+}]
+
+checksums = ['8dd3044767d044aa4ce46de0de562b111c44e5b8b7348e04e665eb1b4f101fe3']
+
+start_dir = 'src'
+configopts = '--force-builddir'
+
+dependencies = [
+    ('Boost', '1.54.0', versionsuffix),
```

Comment:
@SethosII This is a very old version of Boost, while we have existing `bcl2fastq2` that were built on top of very recent versions (e.g. `1.65.1`).

Any reason you're not using the most recent version of Boost for which we have an easyconfig already for `foss/2016b` (which would be `1.63.0`)?

Reply:
@boegel The build fails if I use version `1.61.0` or `1.63.0` and version `1.54.0` is the one that is included in bcl2fastq2 so I chose this version. The error is:

```
/dev/shm/bcl2fastq2/2.20.0/foss-2016b-Python-2.7.12/bcl2fastq/src/cxx/lib/io/Xml.cpp:172:123: error: no matching function for call to ‘xml_writer_make_settings(char, int)’
         boost::property_tree::write_xml(os, treeWithIndexAttributes, boost::property_tree::xml_writer_make_settings(' ', 2));
                                                                                                                           ^
```

It's interesting that you got it to work with `1.65.1`.

Reply:
@SethosII Interesting indeed...

I looked through the Boost release notes, and couldn't pinpoint the breaking change (or bug fix) that could explain this.

So, how do we proceed? Do you prefer sticking to Boost 1.54.0?

Reply:
@boegel Yes, I would prefer to stick with Boost 1.54.0. This also makes me feel better for the time I put into getting this Boost version to compile ;) Hopefully Illumina will eventualy upgrade the dependencies (and fix other stuff, I already told them my complains with this software a while ago).